File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/interfaces/financial/AsyncAttributes.java
package com.emerchantpay.gateway.genesisandroid.api.interfaces.financial;

import com.emerchantpay.gateway.genesisandroid.api.util.RequestBuilder;

public interface AsyncAttributes {

    RequestBuilder requestBuilder = new RequestBuilder("");

    // Async Params
    default AsyncAttributes setReturnSuccessUrl(String successUrl) {
        requestBuilder.addElement("return_success_url", successUrl);
        return this;
    }

    default AsyncAttributes setReturnFailureUrl(String failureUrl) {
        requestBuilder.addElement("return_failure_url", failureUrl);
        return this;
    }

    default RequestBuilder buildAsyncParams() {
        return requestBuilder;
    }
}

File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/interfaces/financial/AsyncAttributes.kt
package com.emerchantpay.gateway.genesisandroid.api.interfaces.financial

import com.emerchantpay.gateway.genesisandroid.api.util.RequestBuilder

interface AsyncAttributes {

    // Async Params
    fun setReturnSuccessUrl(successUrl: String): AsyncAttributes {
        requestBuilder.addElement("return_success_url", successUrl)
        return this
    }

    fun setReturnFailureUrl(failureUrl: String): AsyncAttributes {
        requestBuilder.addElement("return_failure_url", failureUrl)
        return this
    }

    fun buildAsyncParams(): RequestBuilder {
        return requestBuilder
    }

    companion object {

        val requestBuilder = RequestBuilder("")
    }
}
